src/pip/_internal/utils/misc.py

```diff
@@ -427,10 +427,12 @@ def is_local(path):
 
     If we're not in a virtualenv, all paths are considered "local."
 
+    Caution: this function assumes the head of path has been normalized
```

Comment:
Can we just normalize the head of path here and below?

Reply:
We could, but I think the callers need to do it too anyway, so for the use cases we have it would be redundant.

Reply:
Better redundant and give a consistent interface with no gotchas IMO, but I won't block this on this point.